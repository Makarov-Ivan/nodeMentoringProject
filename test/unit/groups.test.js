/* eslint-disable */
const userValidationService = require('../../src/group/services/validation');

describe('users', () => {
  describe('validation service', () => {
    describe('group object', () => {
      it('should return error, if invalid object is passed', async () => {
        const invaliGroup = {
          id: 'unknown',
          name: 10,
          permissions: 0,
        };
        const validationResult = await userValidationService.group(invaliGroup);
        expect(validationResult).toMatchSnapshot();
      });
      it('should return undefined, if valid object is passed', async () => {
        const validUser = {
          id: 20,
          name: 'test group',
          permissions: ['read'],
        };
        const validationResult = await userValidationService.group(validUser);
        expect(validationResult).toBe(undefined);
      });
    });
    describe('usersToGroup string', () => {
      it('should return error, if invalid object is passed', async () => {
        const invaliBody1 = {
          usersIds: [],
        };
        const invaliQuerry2 = {};
        const invaliQuerry3 = '150';
        const validationResult1 = await userValidationService.usersToGroup(invaliBody1);
        expect(validationResult1).toMatchSnapshot();
        const validationResult2 = await userValidationService.usersToGroup(invaliQuerry2);
        expect(validationResult2).toMatchSnapshot();
        const validationResult3 = await userValidationService.usersToGroup(invaliQuerry3);
        expect(validationResult3).toMatchSnapshot();
      });
      it('should return undefined, if valid object is passed', async () => {
        const validBody = {
          usersIds: ['150'],
        };
        const validationResult = await userValidationService.usersToGroup(validBody);
        expect(validationResult).toBe(undefined);
      });
    });
  });
});
