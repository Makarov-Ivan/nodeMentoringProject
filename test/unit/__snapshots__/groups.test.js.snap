// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users validation service group object should return error, if invalid object is passed 1`] = `
Array [
  "\\"id\\" must be a number",
  "\\"name\\" must be a string",
  "\\"permissions\\" must be an array",
]
`;

exports[`users validation service usersToGroup string should return error, if invalid object is passed 1`] = `
Array [
  "\\"usersIds\\" must contain at least 1 items",
]
`;

exports[`users validation service usersToGroup string should return error, if invalid object is passed 2`] = `
Array [
  "\\"usersIds\\" is required",
]
`;

exports[`users validation service usersToGroup string should return error, if invalid object is passed 3`] = `
Array [
  "\\"value\\" must be of type object",
]
`;
